core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -962,6 +961,10 @@ class TimeSeriesShard(val ref: DatasetRef,
     result.onComplete { resp =>
       assertThreadName(IngestSchedName)
       try {
+        // Mark used blocks as reclaimable even on failure. Even if cassandra write fails or other errors occur,
+        // we cannot leave blocks as not reclaimable and also release the factory back into pool.
+        // Can try out tracking unreclaimed blockMemFactories without releasing, but it needs to be separate PR.
+        blockHolder.markUsedBlocksReclaimable()
```

Comment:
What if the used blocks aren't full yet? They might be reclaimed concurrently while filling up more.

Reply:
This block holder is used only by this flush task - no one else because it is checked out of the pool.
So no one else should be filling these blocks.
Once the task is complete, we accept unused portions as defragmentation cost.

Reply:
The absence of markUsedBlocksReclaimable  on flush failure will result in bug. The released factory will be reused for a different flush task, and these blocks will be marked as reclaimable anyway if that task succeeds (even if the chunks are not flushed). It is simpler to mark them as reclaimable rightaway anyway.

Yes - there is a risk of us reclaiming the chunks before flush. But this change doesnt make it worse.